d3a_api_client/redis_aggregator.py

```diff
@@ -44,13 +44,13 @@ def _connect_to_simulation(self, is_blocking=True):
 
     def _subscribe_to_response_channels(self):
         event_channel = f'external-aggregator/*/{self.aggregator_uuid}/events/all'
-        channel_dict = {"aggregator_response": self._aggregator_response_callback,
-                        event_channel: self._events_callback_dict,
+        channel_dict = {event_channel: self._events_callback_dict,
                         f"external-aggregator/*/{self.aggregator_uuid}/response/batch_commands":
                             self._batch_response,
                         }
 
         self.pubsub.psubscribe(**channel_dict)
+        self.pubsub.subscribe(**{"aggregator_response": self._aggregator_response_callback})
```

Comment:
WRT the discussion above, give this a try, it should work. To complement Hannes' answer from before, we also do this in the `RedisSimulationCommunication` class in d3a-web, therefore it should be possible to do so. 
```suggestion
        channel_dict["aggregator_response"] = self._aggregator_response_callback
        self.pubsub.psubscribe(**channel_dict)
```